lib/page/set-uploads/set-uploads.unit.spec.js

```diff
@@ -0,0 +1,484 @@
+    maxFiles: 3,
+    minFiles: 2
+  })
+
+  const {uploadSlots} = await getUploadInstances(pageInstance, userData)
+
+  t.equal(uploadSlots.length, 2, 'it should not decrease the number of upload slots')
+
+  resetGlobalStubs(userData)
+  t.end()
+})
+
+test('When setting up a file upload which has a file already uploaded', async t => {
+  const {userData} = setupUserDataStubs({
+    files: [{
+      uuid: 'bar',
+      size: 10 * 1024,
+      mimetype: 'image/png',
+      originalname: 'foo.png'
+    }]
+  })
+
+  const pageInstance = getPageInstanceWithUploadComponent()
+
+  const {uploadControl, uploadSummary} = await getUploadInstances(pageInstance, userData)
+
+  t.equal(uploadControl.summaryListHeading, 'Already uploaded one', 'it should set the heading for the summary list of uploaded files')
+
+  t.equal(uploadSummary.length, 1, 'it should add the uploaded file to the summary list')
+  const uploadDetails = uploadSummary[0]
+  const uploadKey = uploadDetails.key
+  t.equal(uploadKey.html, 'foo.png (png, 10KB)', 'it should set the text describing the uploaded file')
+  t.equal(uploadKey.classes, 'fb-summary-list--file-key', 'it should set any classes needed for the text describing the uploaded file')
+
+  const uploadActions = uploadDetails.actions.items
+  t.equal(uploadActions.length, 1, 'it should add the the correct number of actions for the uploaded file')
+  const uploadRemove = uploadActions[0]
+  t.equal(uploadRemove.name, 'removeFile', 'it should set the name of the remove button for the uploaded file')
+  t.equal(uploadRemove.value, 'foo:bar', 'it should set the value of the remove button for the uploaded file')
+  t.equal(uploadRemove.html, 'Remove button', 'it should set the remove button text for the uploaded file')
+  t.equal(uploadRemove.visuallyHiddenText, 'foo.png', 'it should set the name of the file as visually hidden text for the remove button for accessibility reasons')
+  t.equal(uploadRemove.classes, 'fb-action-secondary fb-action--delete', 'it should set any classes required for the remove button text for the uploaded file')
+
+  resetGlobalStubs(userData)
+  t.end()
+})
+
+const setupUserDataStubsWithTypeHints = () => {
+  return setupUserDataStubs({
+    strings: {
+      'fileupload.hint': '{types, or}',
+      'fileupload.hint.timing': ''
+    }
+  })
+}
+
+test('When setting up an upload with an accepts that is a type reference rather than a media-type', async t => {
+  const {userData} = setupUserDataStubsWithTypeHints()
+
+  const pageInstance = getPageInstanceWithUploadComponent({
+    accept: ['bar']
+  })
+
+  const {uploadControl, uploadSlots} = await getUploadInstances(pageInstance, userData)
+  const uploadSlot = uploadSlots[0]
+
+  t.equal(uploadControl.accept, 'image/bar,application/foo', 'it should set resolve and expand any type references')
+  t.equal(uploadSlot.hint, 'Bar', 'it should pass any matching hint for the type to the control')
```

Comment:
This is not a question on the tests, but rather the feature, just wondering the reason we convert `image/bar,application/foo` into `bar`?

Reply:
Explained